packages/fuel-gauge/src/predicate/predicate-populate-witness.test.ts

```diff
@@ -0,0 +1,325 @@
+import { seedTestWallet } from '@fuel-ts/account/test-utils';
+import type { BN, CoinQuantityLike, WalletUnlocked } from 'fuels';
+import {
+  BaseAssetId,
+  Provider,
+  Predicate,
+  FUEL_NETWORK_URL,
+  ScriptTransactionRequest,
+  Wallet,
+  cacheResources,
+} from 'fuels';
+
+import { FuelGaugeProjectsEnum, getFuelGaugeForcProject } from '../../test/fixtures';
+
+/**
+ * @group node
+ */
+describe('Predicate', () => {
+  const assertNumberArtifacts = getFuelGaugeForcProject(
+    FuelGaugeProjectsEnum.PREDICATE_ASSERT_NUMBER
+  );
+  const assertValueArtifacts = getFuelGaugeForcProject(
+    FuelGaugeProjectsEnum.PREDICATE_ASSERT_VALUE
+  );
+
+  describe('Populate Predicate Witness', () => {
+    const UTXOS_AMOUNT = 12;
+
+    let gasPrice: BN;
+    let provider: Provider;
+    let receiver: WalletUnlocked;
+    let predicate1: Predicate<[number]>;
+    let predicate2: Predicate<[boolean]>;
+
+    let wallet1: WalletUnlocked;
+    let wallet2: WalletUnlocked;
+    let wallet3: WalletUnlocked;
+
+    const quantity: CoinQuantityLike[] = [[500, BaseAssetId]];
+    beforeAll(async () => {
+      provider = await Provider.create(FUEL_NETWORK_URL, { cacheUtxo: 1000 });
+      wallet1 = Wallet.generate({ provider });
+      wallet2 = Wallet.generate({ provider });
+      wallet3 = Wallet.generate({ provider });
+      receiver = Wallet.generate({ provider });
+      gasPrice = provider.getGasConfig().minGasPrice;
+      predicate1 = new Predicate<[number]>({
+        bytecode: assertNumberArtifacts.binHexlified,
+        provider,
+        abi: assertNumberArtifacts.abiContents,
+        inputData: [11],
+      });
+
+      predicate2 = new Predicate<[boolean]>({
+        bytecode: assertValueArtifacts.binHexlified,
+        abi: assertValueArtifacts.abiContents,
+        provider,
+        inputData: [true],
+      });
+
+      await seedTestWallet(
+        [wallet1, wallet2, wallet3, predicate1, predicate2],
+        [[120_000, BaseAssetId]],
+        UTXOS_AMOUNT
+      );
+    });
+
+    it('should properly populate predicate data and remove placeholder witness [CASE 1]', async () => {
+      let transactionRequest = new ScriptTransactionRequest({ gasPrice, gasLimit: 2000 });
+      transactionRequest.addCoinOutput(receiver.address, 100, BaseAssetId);
+
+      const predicate1WrongResources = await provider.getResourcesToSpend(
+        predicate1.address,
+        quantity
+      );
+
+      transactionRequest.addResources(predicate1WrongResources); // will add a placeholder witness
+
+      // The request carries 1 placeholder witnesses that was added for the predicate resources
+      expect(transactionRequest.witnesses.length).toBe(1);
```

Comment:
```suggestion
      expect(transactionRequest.witnesses.length).toBe(1);
      expect(transactionRequest.witnesses[-1]).toBeTruthy();
```

Just to check it's specifically the place holder withness.

Reply:
@danielbate All witnesses entries will be zeroed placeholders as the wallets resources were not signed at this point.